proto/caraml/upi/v1/dataset_service.proto

```diff
@@ -0,0 +1,107 @@
+syntax = "proto3";
```

Comment:
This file will be moved to `timber` repository after review. Will not be added via this PR (i.e will be removed).

Reply:
I think the other protos in the PR also belong to `timber` since it's quite related with the `DatasetService`